Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -388,48 +410,13 @@ extension SearchViewController: UITableViewDataSource {
         case .SearchSuggestions:
             return searchEngines.shouldShowSearchSuggestions ? 1 : 0
         case .BookmarksAndHistory:
-            return data.count
+            return data?.count ?? 0
```

Comment:
I usually just said that the data-Cursor couldn't be null. Then it always returns a zero length anyway (and we can put a StatusMessage in it if we want to show something in the UI like "Loading..." or "Oops... Something is busted".